client/state/themes/selectors/is-theme-purchased.ts

```diff
@@ -0,0 +1,27 @@
+import { getSitePurchases } from 'calypso/state/purchases/selectors';
+import type { AppState } from 'calypso/types';
+
+import 'calypso/state/themes/init';
+
+/**
+ * Returns whether the theme has been purchased for the given site. Note that this only works for
+ * premium themes that are added to A8c repos -- marketplace themes need to be checked via a different approach.
```

Comment:
Is there a different selector/link to a method we can specify here in case someone goes looking for the marketplace themes logic?

Reply:
I've updated the doc to refer readers to `isMarketplaceThemeSubscribed()`.